bin/find-fixable-orphans.py

```diff
@@ -0,0 +1,293 @@
+    "--env",
+    default="uat",
+    required=False,
+    help="Environment to run against, eg 'production, 'uat', etc (default: '%(default)s')",
+)
+parser.add_argument(
+    "--pgcli",
+    default="/usr/bin/psql",
+    required=False,
+    help=f"Postgres client path (default: %(default)s)",
+)
+args = parser.parse_args()
+
+# Keep synced with other scripts
+GLOSS_ID_COLUMN = "Gloss ID"
+GLOSS_COLUMN = "Gloss"
+GLOSS_PUBLIC_COLUMN = "Gloss public"
+GLOSS_VIDEO_COLUMN = "Suggested Video key"
+GLOBAL_COLUMN_HEADINGS = [
+    GLOSS_ID_COLUMN,
+    GLOSS_COLUMN,
+    GLOSS_PUBLIC_COLUMN,
+    GLOSS_VIDEO_COLUMN,
+]
+
+# Other globals
+CSV_DELIMITER = ","
+FAKEKEY_PREFIX = "this_is_not_a_key_"
+DATABASE_URL = os.getenv("DATABASE_URL", "")
+PGCLI = args.pgcli
+AWS_S3_BUCKET = f"nzsl-signbank-media-{args.env}"
+
+
+def pg_cli(args_list):
+    try:
+        return subprocess.run(
+            [PGCLI, "-c"] + args_list + [f"{DATABASE_URL}"],
+            env=os.environ,
+            capture_output=True,
+            check=True,
+            text=True,
+        )
+    except subprocess.CalledProcessError as e:
+        print(f"Error: subprocess.run returned code {e.returncode}", file=sys.stderr)
+        print(e.cmd, file=sys.stderr)
+        print(e.stdout, file=sys.stderr)
+        print(e.stderr, file=sys.stderr)
+        exit()
+
+
+# Fake key is a hack to handle FULL JOIN
+def maybe_fakekey(instring):
+    return instring if instring else FAKEKEY_PREFIX + str(uuid4())
+
+
+def filter_fakekey(instring):
+    return "" if instring.startswith(FAKEKEY_PREFIX) else instring
+
+
+# Get the video files info from NZSL Signbank
+def get_nzsl_raw_keys_dict():
+    print(
+        f"Getting raw list of video file info from NZSL Signbank ...",
+        file=sys.stderr,
+    )
+    this_nzsl_raw_keys_dict = {}
+    # Column renaming is for readability
+    # Special delimiter because columns might contain commas
+    result = pg_cli(
+        [
+            "COPY ("
+            "SELECT "
+            "dg.id AS gloss_id, "
+            "dg.idgloss AS gloss_idgloss, "
+            "dg.created_at AS gloss_created_at, "
+            "dg.published AS gloss_public, "
+            "vg.is_public AS video_public, "
+            "vg.id AS video_id, "
+            "vg.videofile AS video_key "
+            "FROM dictionary_gloss AS dg "
+            "FULL JOIN video_glossvideo AS vg ON vg.gloss_id = dg.id"
+            ") TO STDOUT WITH (FORMAT CSV, DELIMITER '|')",
+        ]
+    )
+
+    # Separate the NZSL db columns
+    # Write them to a dictionary, so we can do fast operations
+    for rawl in result.stdout.split("\n"):
+        rawl = rawl.strip()
+        if not rawl:
+            continue
+        [
+            gloss_id,
+            gloss_idgloss,
+            gloss_created_at,
+            gloss_public,
+            video_public,
+            video_id,
+            video_key,
+        ] = rawl.split("|")
+
+        # Hack to handle FULL JOIN
```

Comment:
issue: can you put a comment somewhere explaining what the hack actually is?

(this is probably the best place for it, but I wouldn't be upset if you put it on the global variable or in the function docblock)

Reply:
Absolutely!